src/api/valence-api.js

```diff
@@ -0,0 +1,392 @@
+'use strict';
+
+const FormData = require('form-data');
+const { DryRunId, LEVersion, LPVersion } = require('../constants');
+const ContentFactory = require('../utility/content-factory');
+
+module.exports = class ValenceApi {
+	constructor(
+		valence,
+		isDryRun,
+		fetch = require('node-fetch')
+	) {
+		this._valence = valence;
+		this._isDryRun = isDryRun;
+		this._fetch = fetch;
+	}
+
+	async _getContent(instanceUrl, orgUnit, module = null) {
+		const url = module
+			? new URL(`/d2l/api/le/${LEVersion}/${orgUnit.Identifier}/content/modules/${module.Id || module.id}/structure/`, instanceUrl)
+			: new URL(`/d2l/api/le/${LEVersion}/${orgUnit.Identifier}/content/root/`, instanceUrl);
+		const signedUrl = this._valence.createAuthenticatedUrl(url, 'GET');
+
+		if (this._isDryRun && module && module.id === DryRunId) {
+			return [{
+				Id: DryRunId,
+				Type: 0
+			}];
+		}
+
+		const response = await this._fetch(signedUrl);
+
+		this._assertResponse(response);
+
+		return response.json();
+	}
+
+	async _getQuizzes(instanceUrl, orgUnit) {
+		const url = new URL(`/d2l/api/le/${LEVersion}/${orgUnit.Identifier}/quizzes/`, instanceUrl);
+		const signedUrl = this._valence.createAuthenticatedUrl(url, 'GET');
+
+		const response = await this._fetch(signedUrl);
+
+		if (!response.ok) {
+			throw new Error(response.statusText);
+		}
+
+		return response.json();
+	}
+
+	async assertModule(instanceUrl, orgUnit, module, parentModule = null) {
```

Comment:
I wanted to avoid leaking too much implementation detail, so it seemed reasonable to treat this as a manager for the API which does the thing we need with some helpful abstractions.